### PR TITLE
chore: experimental test server implementation

### PR DESCRIPTION
**Pull request by pavelfeldman**

### CODE REVIEW

File: packages/playwright-test/src/reporters/multiplexer.ts [hunk 1]
```diff
@@ -85,3 +93,9 @@ function wrap(callback: () => void) {
     console.error('Error in reporter', e);
   }
 }
+
+type TestRequest = {
+  project: string;
```
Comment (dgozman) [inline packages/playwright-test/src/reporters/multiplexer.ts:98]: `projectId`?

File: packages/playwright-test/src/dispatcher.ts [hunk 2]
```diff
@@ -17,18 +17,21 @@
...
 import { ManualPromise } from 'playwright-core/lib/utils/manualPromise';
+import { TestTypeImpl } from './testType';
 
 export type TestGroup = {
   workerHash: string;
   requireFile: string;
   repeatEachIndex: number;
   projectId: string;
   tests: TestCase[];
+  testServerTestLine?: number;
```
Comment (dgozman) [inline packages/playwright-test/src/dispatcher.ts:34]: You need `requireFile` + `uiLocation`.

File: packages/playwright-test/src/workerRunner.ts [hunk 3]
```diff
@@ -204,6 +209,46 @@ export class WorkerRunner extends EventEmitter {
     }
   }
 
+  private async _runTestServerRequest(runPayload: RunPayload) {
+    this._runFinished = new ManualPromise<void>();
+    try {
+      await this._loadIfNeeded();
+      const fileSuite = await this._loader.loadTestFile(runPayload.file, 'worker');
+      const suite = this._loader.buildFileSuiteForProject(this._project, fileSuite, this._params.repeatEachIndex, test => {
+        if (test.location.line !== runPayload.testServerTestLine)
```
Comment (dgozman) [inline packages/playwright-test/src/workerRunner.ts:218]: Let's put all test-server-specific code into this filter and remove the copy.

**Review by dgozman — APPROVED** [review]

File: packages/playwright-test/src/dispatcher.ts (hunk 2 above)
Comment (pavelfeldman) [reply packages/playwright-test/src/dispatcher.ts:34]: I'd like to keep it as is as a reminder to the limitation.

**Pull request merged**